src/backend/pagemap.h

```diff
@@ -66,6 +66,28 @@ namespace snmalloc
 
     constexpr FlatPagemap() = default;
 
+    /**
+     * For pagemaps that cover an entire fixed address space, return the size
+     * that they must be.  This allows the caller to allocate the correct
+     * amount of memory to be passed to `init`.  This is not available for
+     * fixed-range pagemaps, whose size depends on dynamic configuration.
+     */
+    template<bool HasBounds = has_bounds>
+    static constexpr std::enable_if_t<!HasBounds, size_t> required_size()
+    {
+      constexpr size_t COVERED_BITS = bits::ADDRESS_BITS - GRANULARITY_BITS;
+      constexpr size_t ENTRIES = bits::one_at_bit(COVERED_BITS);
+      return ENTRIES * sizeof(T);
+    }
+
+    /**
+     * Initialise with pre-allocated memory.
+     */
+    void init(T* address)
```

Comment:
My thinking was that it is best to let the Pagemap report the exact size instead of trying to predict how big it wants it to be, but we can figure it out in a future PR if somebody wants to use it like that.